src/main/java/org/reaktivity/nukleus/oauth/internal/OAuthRealms.java

```diff
@@ -153,15 +114,15 @@ public long lookup(
     public boolean unresolve(
         long authorization)
     {
-        long scope = authorization & SCOPE_MASK;
+        long realmId = authorization & REALM_MASK;
         boolean result;
-        if (Long.bitCount(scope) > 1)
+        if (Long.bitCount(realmId) > 1)
         {
             result = false;
         }
         else
         {
-            result = realmsIdsByName.entrySet().removeIf(e -> (e.getValue().realmBit == scope));
+            result = realmsIdsByName.entrySet().removeIf(e -> e.getValue().realmId == realmId);
         }
         return result;
```

Comment:
Simplify to `return Long.bitCount(realmId) <= 1 && realmsById.entrySet().removeIf(...);` instead.